gtsfm/utils/geometry_comparisons.py

```diff
@@ -40,6 +43,122 @@ def align_rotations(aRi_list: List[Rot3], bRi_list: List[Rot3]) -> List[Rot3]:
     return [aRb.compose(bRi) for bRi in bRi_list]
 
 
+def ransac_align_poses_sim3_ignore_missing(
+    aTi_list_ref: List[Optional[Pose3]],
+    bTi_list_est: List[Optional[Pose3]],
+    num_iters: int = 1000,
+    delete_frac: float = DEFAULT_RANSAC_ALIGNMENT_DELETE_FRAC,
+    verbose: bool = False,
+) -> Tuple[List[Optional[Pose3]], Similarity3]:
+    """Align pose graphs by estimating a Similarity(3) transformation, while accounting for outliers in the pose graph.
+
+    Args:
+        aTi_list_ref: pose graph 1 (reference/target to align to).
+        bTi_list_est: pose graph 2 (to be aligned).
+        num_iters: number of RANSAC iterations to execture.
+        delete_frac: what percent of data to remove when fitting a single hypothesis.
+        verbose: whether to print out information about each iteration.
+
+    Returns:
+        best_aligned_bTi_list_est_full: transformed input poses previously "bTi_list" but now which
+            have the same origin and scale as reference (now living in "a" frame). Rather than
+            representing an aligned subset, this represents the entire aligned pose graph.
+        best_aSb: Similarity(3) object that aligns the two pose graphs (best among all hypotheses).
+    """
+    best_aSb = None
+    best_trans_error = float("inf")
+    best_rot_error = float("inf")
+
+    valid_idxs = [i for i, bTi in enumerate(bTi_list_est) if bTi is not None]
+    num_to_delete = math.ceil(delete_frac * len(valid_idxs))
+    if len(valid_idxs) - num_to_delete < 2:
+        # run without RANSAC! want at least 2 frame pairs for good alignment.
+        aligned_bTi_list_est, aSb = align_poses_sim3_ignore_missing(
+            aTi_list_ref, bTi_list_est
+        )
+        return aligned_bTi_list_est, aSb
+
+    # randomly delete some elements
```

Comment:
Why is this implemented this way? Do we know how many minimum number of points are needed for a uniqueue Sim3 estimate? Then we should select those many points.

Reply:
So we need at least 2 poses to align the pose graphs. Since it's least squares, we would expect that with more than 2, we could get a less noisy estimate, though.

Reply:
I think we need 3, there is a rotational ambiguity if we use two. I am not sure about 3 either, just a guess. Could you check that we have at least those many cameras before finding the Sim3?